crossbeam-epoch/src/lib.rs

```diff
@@ -66,6 +66,68 @@ extern crate core;
 
 extern crate maybe_uninit;
 
+#[cfg(feature = "with-loom")]
+extern crate loom;
+
+#[cfg(feature = "with-loom")]
+pub(crate) mod concurrency {
+    pub(crate) mod cell {
+        pub(crate) use loom::cell::UnsafeCell;
+    }
+    pub(crate) mod sync {
+        pub(crate) mod atomic {
+            pub(crate) use loom::sync::atomic::{AtomicUsize, Ordering};
+            pub(crate) fn fence(ord: Ordering) {
+                if let Ordering::Acquire = ord {
+                } else {
+                    // loom only supports acquire fences at the moment
+                    // let's not panic...
+                }
+                loom::sync::atomic::fence(Ordering::Acquire)
```

Comment:
So this would weaken `atomic::fence(Ordering::SeqCst)`s which crossbeam-epoch uses in some places. I was wondering what kind of effects that would have, however I am not sure how hard it is to trigger it with loom?

Reply:
Yes, it would indeed, but I found no other way to get around it at the moment. I filed https://github.com/tokio-rs/loom/issues/117 for things like this that are currently short-comings of this PR.